controllers/numaresourcesoperator_controller_test.go

```diff
@@ -296,6 +298,83 @@ var _ = Describe("Test NUMAResourcesOperator Reconcile", func() {
 						}
 						Expect(reconciler.Client.Get(context.TODO(), mcp2DSKey, ds)).ToNot(HaveOccurred())
 					})
+
+					When("a NodeGroup is deleted", func() {
+						var thirdLoopResult reconcile.Result
+						BeforeEach(func() {
+							// check reconcile second loop result
+							Expect(secondLoopResult).To(Equal(reconcile.Result{RequeueAfter: 5 * time.Second}))
+
+							// check we have at least two NodeGroups
+							Expect(len(nro.Spec.NodeGroups)).To(BeNumerically(">", 1))
```

Comment:
Can you explicitly create a new NRO resource with two node groups that you will add under the reconcile, just to be sure that deleting the node group will not affect other tests?

Reply:
Deleting NodeGroups will only afffect those test cases under this "When" clause, so I think there would be no collision with other testcases

Reply:
While the above is true, I'm also in favor of an explicit test case.

Reply:
Ok, If both of you agree on that I will change it.
I laid out tests this way to avoid code repetition... I mean, I agree that some repetition is ok in test cases and I usually prefer explicit to implicit but being the sut a kind of state machine I though using hierarchical testing features from Ginkgo to go through the different states and assert proper status on each one.